python/mage/constraint_programming/vrp_cp_solver.py

```diff
@@ -0,0 +1,364 @@
+from abc import ABC, abstractmethod
+from gekko import GEKKO
+from mage.geography import VRPPath, VRPResult, VRPSolver
+from typing import List, Tuple
+import numpy as np
+
+
+class VRPConstraintProgrammingSolver(VRPSolver):
+    """
+    This constraint solver solves the Vehicle Routing Problem with constraint programming using GEKKO.
+    """
+
+    SOURCE_INDEX = -1
+    SINK_INDEX = -2
+
+    def __init__(self, no_vehicles: int, distance_matrix: np.array, depot_index: int):
+        self._model = GEKKO(remote=False)
+
+        self.no_vehicles = no_vehicles
+        self.distance_matrix = distance_matrix
+        self.depot_index = depot_index
+
+        self._edge_chosen_vars = dict()
+        self._time_vars = dict()
```

Comment:
Where is `_time_vars` set? Or it is used only as a container in constraint?

Reply:
```
    def _initialize_location_node(self, node_index: int):
        self._time_vars[node_index] = self._model.Var(value=0, lb=0, integer=False)
```